src/icons/Status.js

```diff
@@ -168,27 +168,25 @@ Loading.propTypes = {
     className: propTypes.string,
 }
 
-export const StatusIconNoDefault = ({
-    error,
-    warning,
-    valid,
-    loading,
-    className,
-}) =>
-    valid ? (
-        <Valid className={className} />
-    ) : warning ? (
-        <Warning className={className} />
-    ) : error ? (
-        <Error className={className} />
-    ) : loading ? (
-        <Loading className={className} />
-    ) : null
-
-StatusIconNoDefault.propTypes = {
-    valid: propTypes.bool,
-    error: propTypes.bool,
-    warning: propTypes.bool,
-    loading: propTypes.bool,
+export const StatusIcon = ({ status, defaultTo: DefaultTo, className }) => {
```

Comment:
Regardless of the flag API discussion, I think that the addition of the `defaultTo` prop and renaming `StatusIconNoDefault` to `StatusIcon` is a good idea. This change seems to be fully unrelated to the flags.

Reply:
Yeah I think we could implement that separately 👍

Reply:
Shall I split up the PR into two? `Add defaultTo prop`, and `Use enums instead of mutually exclusive boolean?s`

Reply:
Not sure... Perhaps see where this discussion goes first?